staging/src/k8s.io/mount-utils/mount_linux_test.go

```diff
@@ -545,3 +550,293 @@ func mountArgsContainOption(t *testing.T, mountArgs []string, option string) boo
+		prepare         func(*testing.T) (string, error)
+		needsRootAccess bool
+	}{
+		"path is not absolute": {
+			expectedErr:        errPathNotAbsolute,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				return "../hello/", nil
+			},
+		},
+		"path is a symlink": {
+			expectedErr:        unix.ELOOP,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				tmpDir, err := os.MkdirTemp("", "symlink")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(tmpDir)
+				})
+
+				f, err := os.CreateTemp(tmpDir, "some")
+				if err != nil {
+					return "", err
+				}
+
+				symlink := filepath.Join(tmpDir, "symlink")
+				err = os.Symlink(f.Name(), symlink)
+				if err != nil {
+					return "", err
+				}
+				return symlink, nil
+			},
+		},
+		"path is a mount point": {
+			expectedErr:        nil,
+			expectedMountPoint: true,
+			prepare: func(t *testing.T) (string, error) {
+				// /proc is a known mount-point and since mount-utils heavily relies on /proc/mounts
+				// it is a good option where we can test for mount points in non-privileged mode.
+				return "/proc", nil
+			},
+		},
+		"path is not a mount point": {
+			expectedErr:        nil,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				source, err := os.MkdirTemp("", "source")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(source)
+				})
+				return source, nil
+			},
+		},
+		"path is a magiclink": {
+			expectedErr:        unix.ELOOP,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				// only /proc/1/exe is valid
+				return "/proc/1/exe", nil
+			},
+		},
+		"path does not exist": {
+			expectedErr:        os.ErrNotExist,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				f, err := os.CreateTemp(os.TempDir(), "unexisting")
+				if err != nil {
+					return "", err
+				}
+				err = os.Remove(f.Name())
+				if err != nil {
+					return "", err
+				}
+				return f.Name(), nil
+			},
+		},
+		"parent directory does not exist": {
+			expectedErr:        os.ErrNotExist,
+			expectedMountPoint: false,
+			prepare: func(t *testing.T) (string, error) {
+				return "/random-path/which/does-not/exist", nil
+			},
+		},
+		"check with a normal bind mount (useful for manual testing for reviewers)": {
+			expectedMountPoint: true,
+			needsRootAccess:    true,
+			prepare: func(t *testing.T) (string, error) {
+				sourceDir, err := os.MkdirTemp("", "source")
+				if err != nil {
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.Remove(sourceDir)
+				})
+
+				mountPath, err := os.MkdirTemp("", "target")
+				if err != nil {
+
+					return "", err
+				}
+				t.Cleanup(func() {
+					os.RemoveAll(mountPath)
+				})
+				mounter := New("")
+				err = mounter.Mount(sourceDir, mountPath, "", []string{"bind"})
```

Comment:
This requires the tests running as root. We avoid these. It means you can't really test with real `openat2()`, it must be always mock. I suggest adding a func with mock implementation into every test in this array instead of separate `openAt2Mock` with heuristics. And since all `openat2()` calls are simulated, IMO the tests don't need to touch anything on the filesystem.

(The hard part is getting all the errors from a real `openat2`, but it seems you have that part covered very well).

Reply:
Ah okay! Would you be able to review: https://github.com/moby/sys/pull/97 and very well maintained. From the conversations in this PR and elsewhere I think that is more desirable. 

It also simplifies a lot of the code. thanks for being patient.